feat: add how-to-guide for Dataform (#1958)
kestra-io/docs: content/docs/15.how-to-guides/dataform.md

### Using Github repository

Here is how you can pull an existing project from a GitHub repository and run it with DataformCLI task:

```yaml
id: dataform
namespace: company.team
tasks:
  - id: wdir
    type: io.kestra.plugin.core.flow.WorkingDirectory
    tasks:
      - id: clone_repo
        type: io.kestra.plugin.git.Clone
        url: https://github.com/dataform-co/dataform-example-project-bigquery

      - id: transform
        type: io.kestra.plugin.dataform.cli.DataformCLI
        beforeCommands:
          - npm install @dataform/core
          - dataform compile
        env:
          GOOGLE_APPLICATION_CREDENTIALS: "sa.json"
        inputFiles:
          sa.json: "{{ secret('GCP_SERVICE_ACCOUNT_JSON') }}"
          .df-credentials.json: |
            {
              "projectId": "<gcp-project-id>",
              "location": "us"
            }
        commands:
          - dataform run --dry-run
```

The `clone_repo` task pulls the repository with Dataform project, and the `transform` task executes the Dataform project.

### Using Dataform project creation in Kestra

We will first create the Kestra flow and save it. The flow contains the following tasks:

> 1. HTTP Download task that downloads the orders.csv file using HTTP URL.

Reviewer: ```suggestion
1. HTTP Download task that downloads the `orders.csv` file using HTTP URL.
```

> 2. BigQuery CreateTable task that creates the `orders` table in `ecommerce` dataset.

Reviewer: ```suggestion
2. BigQuery CreateTable task that creates the `orders` table in the `ecommerce` dataset.
```

> 3. BoigQuery Load task that loads the `orders.csv` contents into BigQuery `orders` table.

Reviewer: ```suggestion
3. BigQuery Load task that loads the `orders.csv` contents into the BigQuery `orders` table.
```

> 4. DataformCLI task that will run the Dataform project that we will create later using NamespaceFiles. The project creates the `stg_orders` BigQuery view based on the `orders` BigQuery table.

Reviewer: ```suggestion
4. DataformCLI task that will run the Dataform project, that we will create later using Namespace Files. The project creates the `stg_orders` BigQuery view based on the `orders` BigQuery table.
```

        - name: quantity
          type: INT64
        - name: total
          type: FLOAT64

  - id: load_orders_table
    type: io.kestra.plugin.gcp.bigquery.Load
    description: Load orders table with data from orders.csv
    from: "{{ outputs.orders_http_download.uri }}"
    projectId: <gcp-project-id>
    serviceAccount: "{{ secret('GCP_SERVICE_ACCOUNT_JSON') }}"
    destinationTable: "<gcp-project-id>.ecommerce.orders"
    format: CSV
    csvOptions:
      fieldDelimiter: ","
      skipLeadingRows: 1

  - id: dataform_cli
    type: io.kestra.plugin.dataform.cli.DataformCLI
    beforeCommands:
      - npm install @dataform/core
      - dataform compile
    namespaceFiles:
      enabled: true
    env:
      GOOGLE_APPLICATION_CREDENTIALS: "sa.json"
    inputFiles:
      sa.json: "{{ secret('GCP_SERVICE_ACCOUNT_JSON') }}"
      .df-credentials.json: |
        {
          "projectId": "<gcp-project-id>",
          "location": "us"
        }
    commands:
      - dataform run
```

Once the flow is saved, navigate to the Editor, and create a file `package.json` with the contents:

```json
{
    "dependencies": {
        "@dataform/core": "2.3.0"
    }
}
```

> Note that this file is not required for the Kestra run as we will be installing this dependency using `beforeCommands`. This is however required if you choose to push the namespace files to Github repository so that you can run the project seamlessly in other ways.

Reviewer: Can you clarify the Kestra run part. Do you mean execution here?

```suggestion
Note that this file is not required for the Kestra run as we will be installing this dependency using `beforeCommands`. This is however required if you choose to push the namespace files to GitHub repository so that you can run the project seamlessly in other ways.
```

> Note that this file is not required for the Kestra run as we will be installing this dependency using `beforeCommands`. This is however required if you choose to push the namespace files to GitHub repository so that you can run the project seamlessly in other ways.

Author: yes

> Note that this file is not required for the Kestra run as we will be installing this dependency using `beforeCommands`. This is however required if you choose to push the namespace files to GitHub repository so that you can run the project seamlessly in other ways.

Author: will change it to execution

Next, we will create `dataform.json`.

```json
{
  "warehouse": "bigquery",
  "defaultSchema": "ecommerce",
  "defaultDatabase": "<default-database>",
  "defaultLocation": "us"
}
```

Most often, the `database` is same as the GCP project ID.

Let us now create a folder `definitions`. In this folder create a file `orders.sqlx`. This file will define the `orders` table as the source table. The contents of the `orders.sqlx` file will be:

```
config {
  type: "declaration",
  database: "<database>",
  schema: "ecommerce",
  name: "orders",
  description: "raw orders table"
}
```

Next, we will create the `stg_orders.sqlx` file under the `definitions` folder. This file will define the `stg_orders` view that we want to generate using Dataform. The file contents will be:

```
config {
  type: "view",  // Specify whether this model will create a table or a view
  schema: "ecommerce",
  database: "<database>"
}

select
  order_id,
  customer_name,
  customer_email,
  product_id,
  price,
  quantity,
  total
from
  ${ref("orders")}
```

> Thats it! We are now ready to run the flow.

Reviewer: ```suggestion
That's it! We are now ready to run the flow.
```

> Once the flow runs successfully, you can go to BigQuery console, and ensure that the view `stg_orders` has been created.

Reviewer: ```suggestion
Once the flow runs successfully, you can go to the BigQuery console, and ensure that the view `stg_orders` has been created.
```
